Lecture20(Munchkin)/ManchkinGame_HW/DeckSelectionHelper.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include <vector>
+#include <algorithm>
+#include <random>
+
+template <typename T>
+class DeckSelectionHelper
+{
+public:
+    DeckSelectionHelper(const std::vector<T*>& database) : m_Database(database), m_randomEngine(std::random_device{}())
+    {
+        reset();
+    }
+
+    T* generate()
+    {
+        if (m_Used.empty())
+        {
+            reset();
+        }
+
+        T* result = m_Used.back();
```

Comment:
But why is it called `Used`? This is the place where get new cards, it's more like `available`, or `cardDeck` or similar. Not used)

Reply:
Okay, I can see why you called it `used`, it's just a bit counterintuitive in context of generate function)